source/ambassador_host_test.go

```diff
@@ -17,26 +17,562 @@ limitations under the License.
+			loadBalancer: fakeAmbassadorLoadBalancerService{
+				ips:       []string{"8.8.8.8"},
+				hostnames: []string{"lb.com"},
+				name:      "emissary",
+				namespace: "emissary-ingress",
+			},
+			ambassadorHostItems: []fakeAmbassadorHost{
+				{
+					name:      "fake1",
+					namespace: "testing1",
+					hostname:  "fake1.org",
+					annotations: map[string]string{
+						"external-dns.ambassador-service": "emissary-ingress/emissary",
+						"kubernetes.io/ingress.class":     "external-ingress",
+					},
+					labels: map[string]string{
+						"kubernetes.io/ingress.class": "external-ingress",
+					},
+				},
+			},
+			expected: []*endpoint.Endpoint{
+				{
+					DNSName: "fake1.org",
+					Targets: endpoint.Targets{"8.8.8.8"},
+				},
+				{
+					DNSName: "fake1.org",
+					Targets: endpoint.Targets{"lb.com"},
+				},
+			},
+		},
+		{
+			title:            "valid non matching label filter expression and valid matching annotation filter",
+			targetNamespace:  "",
+			annotationFilter: "kubernetes.io/ingress.class in (external-ingress)",
+			labelSelector: labels.SelectorFromSet(labels.Set{"kubernetes.io/ingress.class": "external-ingress"}),
+			loadBalancer: fakeAmbassadorLoadBalancerService{
+				ips:       []string{"8.8.8.8"},
+				hostnames: []string{"lb.com"},
+				name:      "emissary",
+				namespace: "emissary-ingress",
+			},
+			ambassadorHostItems: []fakeAmbassadorHost{
+				{
+					name:      "fake1",
+					namespace: "testing1",
+					hostname:  "fake1.org",
+					annotations: map[string]string{
+						"external-dns.ambassador-service": "emissary-ingress/emissary",
+						"kubernetes.io/ingress.class":     "external-ingress",
+					},
+					labels: map[string]string{
+						"kubernetes.io/ingress.class": "internal-ingress",
+					},
+				},
+			},
+			expected: []*endpoint.Endpoint{},
+		},
+		{
+			title:            "valid matching label filter expression and non matching annotation filter",
+			targetNamespace:  "",
+			annotationFilter: "kubernetes.io/ingress.class in (external-ingress)",
+			labelSelector: labels.SelectorFromSet(labels.Set{"kubernetes.io/ingress.class": "external-ingress"}),
+			loadBalancer: fakeAmbassadorLoadBalancerService{
+				ips:       []string{"8.8.8.8"},
+				hostnames: []string{"lb.com"},
+				name:      "emissary",
+				namespace: "emissary-ingress",
+			},
+			ambassadorHostItems: []fakeAmbassadorHost{
+				{
+					name:      "fake1",
+					namespace: "testing1",
+					hostname:  "fake1.org",
+					annotations: map[string]string{
+						"external-dns.ambassador-service": "emissary-ingress/emissary",
+						"kubernetes.io/ingress.class":     "internal-ingress",
+					},
+					labels: map[string]string{
+						"kubernetes.io/ingress.class": "external-ingress",
+					},
+				},
+			},
+			expected: []*endpoint.Endpoint{},
+		},
+	} {
+		ti := ti
+		t.Run(ti.title, func(t *testing.T) {
+			// Create a slice of Ambassador Hosts
+			ambassadorHosts := make([]*ambassador.Host, 0)
```

Comment:
Not a big deal at all but I've seen 3 different ways of creating a slice of ambassodor hosts in this PR:
```
var ambassadorHosts []*ambassador.Host
---
filteredList := []*ambassador.Host{}
---
ambassadorHosts := make([]*ambassador.Host, 0)
```
:)

Reply:
Sorry about that copy and paste issues. Thanks for picking that up for me totally missed it

Reply:
I have cleaned this up in https://github.com/kubernetes-sigs/external-dns/pull/2633/commits/cfe7728744c41494c02948fc7da18558f02c3cde decided to use the short declaration operator